Dockerfile.dev

```diff
@@ -0,0 +1,4 @@
+FROM hashicorp/consul-dataplane as cache
```

Comment:
Should we pin this to the latest minor line of tags so we're not relying on `latest` (which might get stale or accidentally overwritten w/ some random version)?

Reply:
IMHO, the `latest` usage is intentional. Since the dockerfile dependencies never really change between releases, it seems safe to use the newest possible version with binaries replaced. My thinking is that in the work case you build 1.0.0 with a newer baser image and dependency like `go-discover`, which should be a safe operation. Also, it's one less thing to update at release time :) 

Open to other cases I haven't thought about though.

Reply:
Ah, TIL [CRT checks](https://go.hashi.co/ui/edit/crt-latest-docker) for the latest minor before updating the `latest` tag 👍🏻 SGTM!